doc/src/guides/solving/solve-numerically.md

```diff
@@ -0,0 +1,216 @@
+
+```py
+>>> from sympy import Symbol, nsolve
+>>> x1 = Symbol('x1')
+>>> x2 = Symbol('x2')
+>>> f1 = 3 * x1**2 - 2 * x2**2 - 1
+>>> f2 = x1**2 - 2 * x1 + x2**2 + 2 * x2 - 8
+>>> print(nsolve((f1, f2), (x1, x2), (-1, 1), prec=25))
+Matrix([[-1.192873099352460791205211], [1.278444111699106966687122]])
+```
+
+### Create a function that can be solved with SciPy
+
+As noted above, SymPy focuses on symbolic computation and is not optimized for
+numerical calculations. A recommend use case for SymPy in a workflow is as given
+below:
+- use SymPy to generate (by symbolically simplifying or solving an equation) the
+  mathematical expression
+- use a numerical library such as SciPy to generate numerical solutions
+
+```py
+>>> from sympy import simplify, cos, sin
+>>> from sympy.abc import x, y
+>>> expr = cos(x * (x + x**2)/(x*sin(y)**2 + x*cos(y)**2 + x))
+>>> simplify(expr)
+cos(x*(x + 1)/2)
+>>> from sympy.utilities.lambdify import implemented_function
+>>> from sympy import lambdify
+>>> f = implemented_function('f', lambda x: cos(x*(x + 1)/2))
+>>> lam_f = lambdify(x, f(x))
+>>> from scipy import optimize
+>>> sol = optimize.root_scalar(lam_f, bracket=[0, 2], method='brentq')
+>>> sol.root
+1.3416277185114782
+```
+
+## Use the solution result
+
+### Substitute the result into an expression
+
+The best practice is to use {func}`sympy.core.evalf` to to substitute numerical
+values into expressions. Using [`subs`](sympy.core.basic.Basic.subs) can give
+incorrect result due to precision errors:
+
+```py
+>>> from sympy import cos, nsolve, Symbol
+>>> x = Symbol('x')
+>>> f = cos(x) - x
+>>> x_value = nsolve(f, x, 1); x_value
+0.739085133215161
+>>> y = Symbol('y')
+>>> z = Symbol('z')
+>>> values = {x: x_value, y: 1e16, z: 1e16}
+>>> (x + y - z).evalf(subs=values) # Gives correct result
+0.739085133215161
+>>> (x + y - z).subs(values) # Gives incorrect result due to precision errors
+0
+```
+
+You can also leave some symbols as variables:
+
+```py
+>>> from sympy import cos, nsolve, Symbol
+>>> x = Symbol('x')
+>>> f = cos(x) - x
+>>> x_value = nsolve(f, x, 1); x_value
+0.739085133215161
+>>> y = Symbol('y')
+>>> z = Symbol('z')
+>>> g = x * y**2
+>>> values = {x: x_value, y: 1}
+>>> (x + y - z).evalf(subs=values)
+1.73908513321516 - z
+```
+
+## Not all equations can be solved
+
+{func}`~.nsolve` is a numerical solving function, so it can often provide a
+solution for equations which cannot be solved algebraically.
+
+### Equations with no solution
+
+Some equations have no solution, in which case SymPy may return an error. For
+example, the equation $exp(x) = 0$ has no solution:
+
+```py
+>>> from sympy import nsolve, exp
+>>> from sympy.abc import x
+>>> nsolve(exp(x), x, 1, prec=20)
+Traceback (most recent call last):
+...
+ValueError: Could not find root within given tolerance. (5.4877893607115270300540019e-18 > 1.6543612251060553497428174e-24)
+Try another starting point or tweak arguments.
+```
```

Comment:
On the other hand, this function gets arbitrarily close to 0 and the default solver will report a false solution because at that value the function is very small:
```python
>>> nsolve(exp(x), x, -1)
-21.8720030830002
```
And currently the bisect routine does not check the sign before checking the value and does not flag this case:
```python
>>> nsolve(exp(x),(-22,-21), solver='bisect')
-22.0000000000000
>>> sign(exp(-22)*exp(-21))
1
>>> nsolve(x**2-1,x,(2,3),solver='bisect')
Traceback (most recent call last):
...
ValueError: Could not find root within given tolerance. (9.0000000000000000052 > 2.16840434497100886801e-19)
Try another starting point or tweak arguments.
```

Reply:
Yeah, haven't found a great example for this--each has some wrinkle to it.